test/tubes/sock_test.rb

```diff
@@ -9,18 +9,17 @@
 class SockTest < MiniTest::Test
   include ::Pwnlib::Tubes
   ECHO_FILE = File.expand_path('../data/echo.rb', __dir__)
-  BIND_PORT = 31_337
 
-  def popen_echo(data)
-    Open3.popen2("bundle exec ruby #{ECHO_FILE} #{BIND_PORT}") do |_i, o, _t|
+  def popen_echo(data, port)
```

Comment:
I think it's better to let it request a free port instead of fixed one

Reply:
or at least use an auto-incremental value in this method

Reply:
But it still might have race-condition.. ignore the above comment unless I have a better idea

Reply:
How about this:
let `data/echo.rb` bind a free port automatically and print the port in first line
i.e.
```ruby
Open3.popen2("bundle exec ruby #{ECHO_FILE} #{port}") do |_i, o, _t|
  port = o.gets.to_i
  yield Sock.new('127.0.0.1', port), data, o
end
```